src/common/cockpitwebresponse.c

```diff
@@ -1709,3 +1716,9 @@ cockpit_web_response_content_type (const gchar *path)
 
   return NULL;
 }
+
+void
+cockpit_web_response_set_head_only (CockpitWebResponse *response)
```

Comment:
I mentioned that in a brief discussion with Peter in #6202. I initially thought about that, but keeping track of the ownership of the string is a (small) burden which is easy to avoid with a boolean, especially as we don't want to support POST/PUT/DELETE etc. But I can look into that, sure. I'll first work on the missing tests though.